playbooks/ocp-install-preparation.yml

```diff
@@ -1,64 +1,48 @@
 ---
+- name: OCP | Install OpenShift Ansible packages on installer
+  hosts: localhost
+  vars_files:
+    - '../defaults/main.yml'
+  become: yes
+  tasks:
+    - name: Install OpenShift Ansible packages
+      package:
+        name: "openshift-ansible*"
+        state: latest
+      tags:
+        installer-prep
+      when: ansible_facts['distribution'] == "RedHat"
+
 - name: OCP | Host Registration
   hosts: all
+  become: yes
+  vars_files:
+    - '../defaults/main.yml'
   tasks:
     - name: Disable all repos
-      command: "subscription-manager repos --disable='*'"
+      rhsm_repository:
+        name: '*'
+        state: disabled
 
     - name: Enable Required repos
-      command: "subscription-manager repos --enable='{{ item }}'"
-      with_items: "{{ ocp_repositories }}"
+      rhsm_repository:
+        name: "{{ item }}"
```

Comment:
as per https://docs.ansible.com/ansible/latest/modules/rhsm_repository_module.html you can pass `name` a list of names with `purge: true` and it will ensure only those repos are enabled idompotently. So just pass `{{ ocp_repositories[ocp_deployment_version | string]  }}` here.